dpnp/dpnp_iface_indexing.py

```diff
@@ -1344,31 +1343,125 @@ def select(condlist, choicelist, default=0):
+
+    Examples
+    --------
+    >>> import dpnp as np
+
+    Beginning with an array of integers from 0 to 5 (inclusive),
+    elements less than ``3`` are negated, elements greater than ``3``
+    are squared, and elements not meeting either of these conditions
+    (exactly ``3``) are replaced with a `default` value of ``42``.
+
+    >>> x = np.arange(6)
+    >>> condlist = [x<3, x>3]
+    >>> choicelist = [x, x**2]
+    >>> np.select(condlist, choicelist, 42)
+    array([ 0,  1,  2, 42, 16, 25])
+
+    When multiple conditions are satisfied, the first one encountered in
+    `condlist` is used.
+
+    >>> condlist = [x<=4, x>3]
+    >>> choicelist = [x, x**2]
+    >>> np.select(condlist, choicelist, 55)
+    array([ 0,  1,  2,  3,  4, 25])
+
     """
 
-    if not use_origin_backend():
-        if not isinstance(condlist, list):
-            pass
-        elif not isinstance(choicelist, list):
-            pass
-        elif len(condlist) != len(choicelist):
-            pass
-        else:
-            val = True
-            size_ = condlist[0].size
-            for cond, choice in zip(condlist, choicelist):
-                if cond.size != size_ or choice.size != size_:
-                    val = False
-            if not val:
-                pass
-            else:
-                return dpnp_select(condlist, choicelist, default).get_pyobj()
+    if len(condlist) != len(choicelist):
+        raise ValueError(
+            "list of cases must be same length as list of conditions"
+        )
+
+    if len(condlist) == 0:
+        raise ValueError("select with an empty condition list is not possible")
+
+    dpnp.check_supported_arrays_type(*condlist)
+    dpnp.check_supported_arrays_type(*choicelist)
+    dpnp.check_supported_arrays_type(
+        default, scalar_type=True, all_scalars=True
+    )
+
+    if dpnp.isscalar(default):
+        usm_type_alloc, sycl_queue_alloc = get_usm_allocations(
+            condlist + choicelist
+        )
+        dtype = dpnp.result_type(*choicelist)
+        default = dpnp.asarray(
+            default,
+            dtype=dtype,
+            usm_type=usm_type_alloc,
+            sycl_queue=sycl_queue_alloc,
+        )
+        choicelist.append(default)
+    else:
+        choicelist.append(default)
+        usm_type_alloc, sycl_queue_alloc = get_usm_allocations(
+            condlist + choicelist
+        )
+        dtype = dpnp.result_type(*choicelist)
+
+    for i, cond in enumerate(condlist):
+        if cond.dtype.type is not dpnp.bool:
+            raise TypeError(
+                f"invalid entry {i} in condlist: should be boolean ndarray"
+            )
+
+    # Convert conditions to arrays and broadcast conditions and choices
+    # as the shape is needed for the result
+    condlist = dpnp.broadcast_arrays(*condlist)
+    choicelist = dpnp.broadcast_arrays(*choicelist)
+
+    result_shape = dpnp.broadcast_arrays(condlist[0], choicelist[0])[0].shape
+
+    result = dpnp.full(
```

Comment:
What if to use `full_like` instead? I guess there will be less number of keywords needs to be specified then.